cde-core/src/pt/webdetails/cdf/dd/model/inst/writer/cdfrunjs/dashboard/amd/PentahoCdfRunJsDashboardWriteContext.java

```diff
@@ -50,5 +55,16 @@ public String replaceTokens( String content ) {
       //build the system resource links
       .replaceAll( ABS_SYS_RES_TAG, "/" + getSystemDir() + "/" + getPluginId( path ) + "$1" )
       .replaceAll( REL_SYS_RES_TAG, path + "$1" );
+
+    return result;
+  }
+
+  // method is "protected" in order to being accessible from tests
+  protected String getRoot() {
```

Comment:
Since both the amd and legacy version of `PentahoCdfRunJsDashboardWriteContext` extend from `CdfRunJsDashboardWriteContext`, can you move this method to that class and remove it from both that extend from it?